airflow/providers/amazon/aws/operators/ecs.py

```diff
@@ -140,10 +141,13 @@ def get_last_log_messages(self, number_messages) -> list:
 
     def get_last_log_message(self) -> Optional[str]:
         try:
-            return self.get_last_log_messages(1)[0]
+            return next(self._get_log_events(start_from_head=False))['message']
         except IndexError:
             return None
 
+    def get_all_log_messages(self) -> Optional[list]:
+        return [log['message'] for log in deque(self._get_log_events())]
```

Comment:
Using `deque` here is useless here since you end up holding all the messages in memory anyway.

```suggestion
    def get_all_log_messages(self) -> Optional[list]:
        return [log['message'] for log in self._get_log_events()]
```